packages/flutter_tools/lib/src/fuchsia/fuchsia_device.dart

```diff
@@ -595,6 +596,33 @@ class FuchsiaDevice extends Device {
   @override
   void clearLogs() {}
 
+  @override
+  VMServiceDiscoveryForAttach getVMServiceDiscoveryForAttach({
+    String? appId,
+    String? fuchsiaModule,
+    int? filterDevicePort,
+    int? expectedHostPort,
+    required bool ipv6,
+    required Logger logger,
+  }) {
+    if (fuchsiaModule == null) {
+      throwToolExit("'--module' is required for attaching to a Fuchsia device");
+    }
+    if (expectedHostPort != null) {
+      throwToolExit("'--host-vmservice-port' is not supported when attaching to a Fuchsia device");
+    }
+    FuchsiaIsolateDiscoveryProtocol? isolateDiscoveryProtocol;
+    try {
+      isolateDiscoveryProtocol = getIsolateDiscoveryProtocol(fuchsiaModule);
+      return FuchsiaIsolateVMServiceDiscoveryForAttach(isolateDiscoveryProtocol);
+    } on Exception {
+      isolateDiscoveryProtocol?.dispose();
+      final List<ForwardedPort> ports = portForwarder.forwardedPorts.toList();
+      ports.forEach(portForwarder.unforward);
```

Comment:
So previously, it was
https://github.com/flutter/flutter/blob/137cb4be670d48f3d9b45a5a6c6cd50df488b4a7/packages/flutter_tools/lib/src/commands/attach.dart#L309-L311

Does putting it in a `forEach` wait for each?